qiskit/algorithms/eigensolvers/numpy_eigensolver.py

```diff
@@ -0,0 +1,297 @@
+
+    def _get_energies(
+        self,
+        operator: BaseOperator | PauliSumOp,
+        aux_operators: ListOrDict[BaseOperator | PauliSumOp] | None,
+    ) -> None:
+        if self._ret.eigenvalues is None or self._ret.eigenstates is None:
+            self._solve(operator)
+
+        if aux_operators is not None:
+            aux_op_vals = []
+            for i in range(self._k):
+                aux_op_vals.append(
+                    self._eval_aux_operators(aux_operators, self._ret.eigenstates[i])
+                )
+            self._ret.aux_operator_eigenvalues = aux_op_vals
+
+    @staticmethod
+    def _eval_aux_operators(
+        aux_operators: ListOrDict[BaseOperator | PauliSumOp],
+        wavefn: np.ndarray,
+        threshold: float = 1e-12,
+    ) -> ListOrDict[tuple[complex, complex]]:
+
+        values: ListOrDict[tuple[complex, complex]]
+
+        # As a list, aux_operators can contain None operators for which None values are returned.
+        # As a dict, the None operators in aux_operators have been dropped in compute_eigenvalues.
+        if isinstance(aux_operators, list):
+            values = [None] * len(aux_operators)
+            key_op_iterator = enumerate(aux_operators)
+        else:
+            values = {}
+            key_op_iterator = aux_operators.items()
+        for key, operator in key_op_iterator:
+            if operator is None:
+                continue
+            value = 0.0
+            if operator.coeff != 0:
+                mat = operator.to_spmatrix()
+                # Terra doesn't support sparse yet, so do the matmul directly if so
+                # This is necessary for the particle_hole and other chemistry tests because the
+                # pauli conversions are 2^12th large and will OOM error if not sparse.
+                if isinstance(mat, scisparse.spmatrix):
+                    value = mat.dot(wavefn).dot(np.conj(wavefn))
+                else:
+                    if isinstance(operator, PauliSumOp):
+                        value = (
+                            Statevector(wavefn).expectation_value(operator.primitive)
+                            * operator.coeff
+                        )
+                    else:
+                        value = Statevector(wavefn).expectation_value(operator)
+                value = value if np.abs(value) > threshold else 0.0
+            # The value gets wrapped into a tuple: (mean, standard deviation).
+            # Since this is an exact computation, the standard deviation is known to be zero.
+            values[key] = (value, 0.0)
+        return values
+
+    def compute_eigenvalues(
+        self,
+        operator: BaseOperator | PauliSumOp,
+        aux_operators: ListOrDict[BaseOperator | PauliSumOp] | None = None,
+    ) -> NumPyEigensolverResult:
+
+        super().compute_eigenvalues(operator, aux_operators)
+
+        if operator is None:
+            raise AlgorithmError("Operator was never provided")
+
```

Comment:
I am not sure this is needed anymore. None is not a valid value for this method usually we don't police this. I imagine in the existing code this has hung around from a time we could supply an operator on the constructor or later with both defaulting to None so you could pass it on either we needed such a check to make sure you had done so.